packages/blade/src/components/Dropdown/_decisions/decisions.md

```diff
@@ -0,0 +1,697 @@
+  <ActionListItem
+    title="Item Value"
+    desciption="Item Description"
+    leading={StarIcon} // Can be IconComponent type or Image URL
+    trailing="⌘ + Space"
+  />
+  <ActionListDivider />
+  <ActionListFooter
+    title="Footer Title"
+    description="Footer Description"
+    leading={DocIcon}
+    trailing={{
+      primary: {
+        onClick: () => {},
+        text: 'Primary Action',
+      },
+    }}
+    // OR
+    trailing={ArrowRightIcon}
+    // OR
+    trailing={{
+      primary: {
+        onClick: () => {
+          console.log('Feedback Yes Clicked');
+        },
+        icon: CheckIcon,
+      },
+      secondary: {
+        onClick: () => {
+          console.log('Feedback No Clicked');
+        },
+        icon: CrossIcon,
+      },
+    }}
+  />
+</ActionList>
+```
+
+</details>
+
+### `ActionList` Parent Component
+
+Wrapper for all ActionList compound components. Should handle the logic of selections.
+
+#### API
+
+Usage
+
+```jsx
+<ActionList>{/* Child components */}</ActionList>
+```
+
+Props
+
+```ts
+type ActionListProps = {
+  children: React.ReactNode;
+
+  /**
+   * Renders simple menu or checkboxes depending on the variant
+   *
+   * @default single
+   */
+  variant?: 'single' | 'multiple';
+
+  /** Callback that gets called when item is checked/clicked */
+  onChange?: () => void;
+
+  /**
+   * Changes the background surface level color
+   *
+   * @default 2
+   */
+  surfaceLevel: 2 | 3;
+};
+```
+
+### `ActionListItem`
+
+![](2022-11-29-17-29-01.png)
+
+#### Design
+
+- [Figma - ActionListItem / Body](https://www.figma.com/file/sAdplk2uYnI2ILnDKUxycW/Blade---Banking-Dark?node-id=13136%3A378703&t=RoEboUS3qWdStKK6-0)
+
+#### API
+
+Usage
+
+```jsx
+<ActionListItem
+  title="Accept API"
+  value="accept"
+  desciption="Accept this API"
+  leading={<ActionListItemIcon icon={StarIcon} />}
+  intent="negative"
+  trailing={<ActionListItemText>⌘ + Space + A</ActionListItemText>}
+  isDefaultSelected={false}
+  onClick={() => {
+    console.log('Item clicked');
+  }}
+/>
+```
+
+Props
+
+```ts
+type ActionListItemProps = {
+  title: string;
+
+  description: string;
+
+  /** Value that we use in JS form. Similar to `value` in `Checkbox` */
+  value: string;
+
+  /**
+   * Leading Icon or assets. Accepts JSX Component
+   *
+   * Should not be allowed when ActionList variant is multiple select
+   */
+  leading?: React.ReactNode;
+
+  /**
+   * Trailing Assets. Should ideally be ActionListItemText component for now
+   */
+  trailing?: React.ReactNode;
+
+  /**
+   * Changes item color to red
+   *
+   * @default neutral
+   */
+  intent?: 'negative' | 'neutral';
+
+  /** Click event on item */
+  onClick?: () => void;
+
+  /**
+   * Set true if you want item to be selected by default
+   *
+   * Should not be allowed when intent is set to negative
+   */
+  isDefaultSelected?: boolean;
+
+  /** URL. Should render item as `a` when this is passed */
+  href?: string;
+};
+```
+
+### `ActionListSection`
```

Comment:
i hope to make sure all these subcomponents will be used under the context of allowed parents we will be adding context?

Reply:
yup we'll put some validations on these components